File: Bindings~/java/src/main/java/com/alttester/Commands/InputActions/AltKeysUp.java
/*
    Copyright(C) 2026 Altom Consulting
*/

package com.alttester.Commands.InputActions;

import com.alttester.Commands.AltBaseCommand;
import com.alttester.IMessageHandler;

public class AltKeysUp extends AltBaseCommand {

  private AltKeysUpParams params;

  public AltKeysUp(IMessageHandler messageHandler, AltKeysUpParams params) {
    super(messageHandler);
    this.params = params;
  }

  public void Execute() {
    SendCommand(params);
    recvall(params, String.class);
  }
}

File: Assets/AltTester/Runtime/AltDriver/Commands/InputActions/AltKeysUp.cs
/*
    Copyright(C) 2026 Altom Consulting
*/

namespace AltTester.AltTesterSDK.Driver.Commands
{
    public class AltKeysUp : AltBaseCommand
    {
        AltKeysUpParams cmdParams;

        public AltKeysUp(IDriverCommunication commHandler, AltKeyCode[] keyCodes) : base(commHandler)
        {
            this.cmdParams = new AltKeysUpParams(keyCodes);
        }
        public void Execute()
        {
            CommHandler.Send(cmdParams);
            var data = CommHandler.Recvall<string>(cmdParams);
            ValidateResponse("Ok", data);
        }
    }
}
